client/packages/purchase_orders/src/DetailView/AppBarButtons/AddButton.tsx

```diff
@@ -0,0 +1,94 @@
+import React, { useEffect, useMemo, useState } from 'react';
+import { SplitButton, SplitButtonOption } from '@common/components';
+import { useTranslation } from '@common/intl';
+import { AddFromMasterListButton } from './AddFromMasterListButton';
+import { useNotification, useToggle } from '@common/hooks';
+import { PlusCircleIcon } from '@common/icons';
+import { PurchaseOrderFragment } from '../../api';
+import { PurchaseOrderNodeStatus } from '@common/types';
+
+interface AddButtonProps {
+  purchaseOrder: PurchaseOrderFragment | undefined;
+  onAddItem: (newState: boolean) => void;
+  /** Disable the whole control */
+  disable: boolean;
+  disableAddFromMasterListButton: boolean;
+  disableAddFromInternalOrderButton: boolean;
+}
+
+export const AddButton = ({
+  purchaseOrder,
+  onAddItem,
+  disable,
+  disableAddFromMasterListButton,
+}: AddButtonProps) => {
+  const t = useTranslation();
+  const { info } = useNotification();
+  const masterListModalController = useToggle();
+
+  const options: [SplitButtonOption<string>, SplitButtonOption<string>] =
+    useMemo(
+      () => [
+        {
+          value: 'add-item',
+          label: t('button.add-item'),
+          isDisabled: disable,
+        },
+        {
+          value: 'add-from-master-list',
+          label: t('button.add-from-master-list'),
+          isDisabled: disableAddFromMasterListButton || disable,
+        },
+      ],
+      [disable, disableAddFromMasterListButton]
+    );
+
+  const [selectedOption, setSelectedOption] = useState<
+    SplitButtonOption<string>
+  >(options[0]);
+
+  useEffect(() => {
+    setSelectedOption(options[0]);
+  }, [options]);
+
+  const handleOptionSelection = (option: SplitButtonOption<string>) => {
+    switch (option.value) {
+      case 'add-item':
+        onAddItem(true);
+        break;
+      case 'add-from-master-list':
+        // Mimmicking OG behaviour where purchase orders can be edited when confirmed AND when authorised
+        purchaseOrder?.status !== PurchaseOrderNodeStatus.Finalised
```

Comment:
As per validation in back end, can add lines to PO when it is anything but FINALISED

This is the same as OG purchase order behaviour